lib/index.js

```diff
@@ -94,6 +94,11 @@ Server.prototype.checkRequest = function(req, fn) {
 Server.prototype.serveClient = function(v){
   if (!arguments.length) return this._serveClient;
   this._serveClient = v;
+
+  if(v && !clientSource) {
+      clientSource = read(require.resolve('socket.io-client/socket.io.js'), 'utf-8');
```

Comment:
Wouldn't it make more sense to load the client source in the  [Server's .serve() method](https://github.com/codigi/socket.io/blob/a9c9b3dcc8afe890b406880be7f65906e4b5853a/lib/index.js#L291-L294)?